avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeCache.java

```diff
@@ -296,6 +339,26 @@ private String getSchemaKey(Schema writerSchema, Schema readerSchema) {
         Utils.getSchemaFingerprint(readerSchema));
   }
 
+  /**
+   * A wrapper function that limits the total number of fast de/serilizer calsses generated
+   *
+   * @param runnable The function to build and save fast de/serializer
+   */
+  private void buildFastClassWithNumCheck(Runnable runnable) {
+    try {
+      if (this.generatedSerDesNum.get() < this.generatedFastSerDesLimit) {
+        runnable.run();
+      } else if (this.generatedSerDesNum.get() == this.generatedFastSerDesLimit) {
+        LOGGER.warn("Generated fast serdes classes number hits limit {}", this.generatedFastSerDesLimit);
+      } else if (this.generatedSerDesNum.get() > this.generatedFastSerDesLimit) {
+        LOGGER.debug("Generated serdes number {}, with fast serdes limit set to {}", this.generatedSerDesNum.get(), this.generatedFastSerDesLimit);
+      }
+      this.generatedSerDesNum.incrementAndGet();
+    } catch (Exception e) {
+      LOGGER.error("Fast serdes class generation failed");
```

Comment:
Shall we print out the full error stacktrace here?

Reply:
The exceptions are handled in other functions. So a stack trace here may not provide any useful information.